stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/ReadCharacters.java

```diff
@@ -55,29 +65,81 @@
      */
     private static final int NUMBER_OF_CHARS_INDEX = 0;
 
+    /**
+     * Will be the I/O event handler.
+     */
+    private EventManager eventManager = EventManager.getInstance();
+
+    private static final Logger log = LoggerFactory.getLogger(ReadCharacters.class);
+
+    /*
+     * Callback method of the read characters response.
+     *
+     * @param result the result returned as the response.
+     * @return the processed event result.
+     *//*
+    private static EventResult readCharactersResponse(EventResult<Boolean, EventContext> result) {
+        BStruct errorStruct;
+        EventContext eventContext = result.getContext();
+        Context context = eventContext.getContext();
+        Throwable error = eventContext.getError();
+        if (null != error) {
+            errorStruct = IOUtils.createError(context, error.getMessage());
+        }
+        Boolean numberOfBytes = result.getResponse();
+        return result;
+    }
```

Comment:
Remove this uncommented code.

Reply:
This needs to be uncommented upon the availability of the async functions